src/vaults/ERC4626/ERC4626.sol

```diff
@@ -27,10 +27,10 @@ abstract contract ERC4626 is IERC4626, Auth, Accounting, Fees, ERC20 {
                                IMMUTABLES
     //////////////////////////////////////////////////////////////*/
 
-	ERC20 public immutable asset;
+	ERC20 public immutable _asset;
```

Comment:
is the point of this refactor to make this variable private?

Reply:
No, this was done so we could create a function called asset(). It's only necessary because the ERC4626 declares that we need a public function called asset which returns the address of underlying token. The only reason is to strict follow the standard, which we can choose not do since a public variable has it's own "getter" function. Just not sure if we call asset() it will return the address of asset.

Reply:
Yes, `ERC20 public immutable asset;` will fulfill the asset() interface, so there is no need to do this refactor.
Generally the only reason you would want to add a custom getter is to make `_asset` private so that inheriting contracts don't accidental overwrite it. In this case, however, this variable is immutable so not overwrite is possible anyway.

Lets revert this refactor to make the code a little cleaner.